src/hashtable/hashtable.h

```diff
@@ -61,6 +64,15 @@ struct hashtable_config {
  * in ad-hoc allocated memory if needed.
  * The struct is aligned to 32 byte to ensure to fit the first half or the second half of a cache-line
  */
+static log_producer_t* hashmap_log_producer;
+static void __attribute__((constructor)) init_hashmap_log(){
```

Comment:
Not sure if it's a good idea having the constructor and destructor in an header:
- using static is to reduce visibility on purpose because the usage is really limited and it's internal, so it shouldn't be in an header
- placing it in an header every C file including the header would end up having its own copy (ie. I am compiling the tests from your branch and the C++ compiler is complaining with a warning caused by the code that it is fine with C because it is actually compiling it for every inclusion)

The same goes for the static log_producer_t and the destructor.

Unless you need it for some specific reason, you can simply move them, as they are, at the beginning of the C file and that's it.

The same goes for the other ones as well.

Reply:
Here some more insights about statics in the headers
https://stackoverflow.com/a/42058453/169278

Reply:
It has to be here because this log producer is used in multiple hashmap source files.

Reply:
Oh, I see, I just checked and you are totally right, it's in use in hashtable_data.c.

The original intent with the "TAG" was to have standard prefix to be able to easily identify the context of the log message, but the hashtable should be seen asa a single component and not a bunch of them, so it's correct to have a single point for the hashtable producer.

The problem is that every single file that will include the header will have it's own producer, having hundred of copies in memory, the hashtable.h header is supposed to be included almost everywhere.

To avoid that moving the contructor/destructor inside hashtable.c and dropping the static attribute from static log_producer_t* hashmap_log_producer; should be enough, in this way the instance of the variable will be shared between the components that will include hashtable.h and only the constructor within hashtable.c will set its value.
Constructor that is even called before the main so it's safe to do.